lib/auth/init.go

```diff
@@ -493,6 +500,182 @@ func migrateLegacyResources(ctx context.Context, cfg InitConfig, asrv *Server) e
 	return nil
 }
 
+const migrationAbortedMessage = "migration to RBAC has aborted because of the backend error, restart teleport to try again"
+
+// migrateOSS performs migration to enable role-based access controls
+// to open source users. It creates a less privileged role 'user'
+// and migrates all users and trusted cluster mappings to it
+// this function can be called multiple times
+// DELETE IN(7.0)
+func migrateOSS(ctx context.Context, asrv *Server) error {
+	if modules.GetModules().BuildType() != modules.BuildOSS {
+		return nil
+	}
+	role := services.NewOSSUserRole()
+	err := asrv.CreateRole(role)
+	createdRoles := 0
+	if err != nil {
+		if !trace.IsAlreadyExists(err) {
+			return trace.Wrap(err, migrationAbortedMessage)
+		}
+		// Role is created, assume that migration has been completed.
+		// To re-run the migration, users can delete the role.
```

Comment:
Have you considered updating the role with `migrate-v6.0` label as a very last step of the migration? This way, even if something below fails, the migration will re-run automatically.

Reply:
I'm not updating existing role here, creating a new one